File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> nMap;

        for(int n = 0; n < nums.size(); n++)
        {
            int diff = target - nums[n];
            if(nMap.find(diff) != nMap.end())
            {
                return {nMap[diff], n};
            }
            nMap[nums[n]] = n;
        }
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        num_map = {}
        for i in range(len(numbers)):
            diff = target - numbers[i]
            if num_map.get(diff) is not None:
                return [num_map[diff], i + 1]
            num_map[numbers[i]] = i + 1